mithril-stm/src/single_signature/signature.rs

```diff
@@ -125,14 +125,21 @@ impl SingleSignature {
         })
     }
 
+    // Visibility will be changed to private.
     /// Compare two `SingleSignature` by their signers' merkle tree indexes.
-    pub fn cmp_stm_sig(&self, other: &Self) -> Ordering {
+    pub fn compare_signer_index(&self, other: &Self) -> Ordering {
         self.signer_index.cmp(&other.signer_index)
     }
 
-    /// Verify a core signature by checking that the lottery was won,
+    /// Will be deprecated. Use `compare_signer_index` instead.
+    pub fn cmp_stm_sig(&self, other: &Self) -> Ordering {
+        SingleSignature::compare_signer_index(self, other)
+    }
```

Comment:
You could do like this:
```suggestion
/// Compare two `SingleSignature` by their signers' merkle tree indexes.
    fn compare_signer_index(&self, other: &Self) -> Ordering {
self.signer_index.cmp(&other.signer_index)
}

    /// Compare two `SingleSignature` by their signers' merkle tree indexes.
    #[deprecated(since = "0.4.X", note = "This function will be removed")]
    pub fn cmp_stm_sig(&self, other: &Self) -> Ordering {
        Self::compare_signer_index(self, other)
    }
```

However, this specific function could be simply removed and its code put in the `Ord` implementation below.